node/test/util/stash_util.js

```diff
@@ -338,6 +338,11 @@ x=E:Fmeta-stash=s;
                     }
                 }
 
+                for (const c of Object.keys(commitMap)) {
```

Comment:
why is this needed?

Reply:
In some cases, the first or second parent might be an existing commit, but the test framework only wants commitMap to contain commits it has never seen, so we have to remove the ones that already existed. I will add a comment to this effect.